src/Sound/Tidal/UI.hs

```diff
@@ -1439,6 +1445,47 @@ _arp name p = arpWith f p
         thumbup xs = concatMap (\x -> [thumb,x]) $ tail xs
           where thumb = head xs
 
+{- | `rolled` plays each note of a chord quickly in order, as opposed to simultaneously; to give a chord a harp-like effect.
+This will played from the lowest note to the highest note of the chord
+@
+rolled $ n "c'maj'4" # s "superpiano"
+@
+
+You can reverse this order by using the `rolled'` function
+
+@
+rolled' $ n "c'maj9" # s "superpiano"
+@
+
+And you can use `rolledBy` or `rolledBy'` to specify the length of the roll. The value in the passed pattern
+is the divisor of the cycle length.
+
+@
+rolledBy "<1 2 4 8 16 3>" $ note "c'maj9" # s "superpiano"
+@
+-}
+
+rolledWith :: ([Event a] -> [EventF (ArcF Time) a]) -> Ratio Integer -> Pattern a -> Pattern a
+rolledWith f t = withEvents aux
+         where aux es = concatMap (steppityIn) (groupBy (\a b -> whole a == whole b) $ sortOn whole (f es))
+               steppityIn xs = mapMaybe (\(n, x) -> shiftIt n (length xs) x) $ enumerate xs
+               shiftIt n d (Event c (Just (Arc s e)) a' v) = do
```

Comment:
This gives a 'Pattern match(es) are non-exhaustive' warning, I think this will break on continuous events (those with whole arc of `Nothing`). Probably adding a catchall of `shiftIt _ _ ev = ev` is enough.

Reply:
I resolved this issue. This will ignore the rolled function for continuous events like
```
n (rolledBy "1" (irand 8) |+ "[0,8, 12]")

-- ~0>1~|n: Note {unNote = 2.0}n
-- ~0>1~|n: Note {unNote = 10.0}n
-- ~0>1~|n: Note {unNote = 14.0}n
```